### PR TITLE
Introduce Target Managers

### LINKED ISSUE #2031: Disconnect from browser occurs every 20-50 page.GoToAsync()
Hello, following code throws an exception about one time per 20-50 navigations. Every time it is occurs on random link at random time.

```c#
  var options = new ConnectOptions();
  options.BrowserWSEndpoint = "ws://127.0.0.1:9222/devtools/browser/dbc9bf45-1b64-4928-a4b3-34e1978237e0b2";

  var browser = await Puppeteer.ConnectAsync(options);
  var page = await browser.NewPageAsync();

  while (true)
  {
      foreach (var website in websites)
      {
          await page.GoToAsync(website, 120_000);
          await Task.Delay(1_000);
      }
  }
```

Most often exception looks like:
`
Navigation failed because browser has disconnected! (Connection failed to process Target.attachedToTarget. Protocol error(Page.setLifecycleEventsEnabled): Target closed. (Target.detachedFromTarget).    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PuppeteerSharp.CDPSession.<SendAsync>d__33.MoveNext() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\CDPSession.cs:line 97
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PuppeteerSharp.FrameManager.<InitializeAsync>d__44.MoveNext() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\FrameManager.cs:line 83
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PuppeteerSharp.FrameManager.<OnAttachedToTargetAsync>d__51.MoveNext() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\FrameManager.cs:line 258
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PuppeteerSharp.FrameManager.<>c__DisplayClass49_0.<<Client_MessageReceived>b__0>d.MoveNext() in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\FrameManager.cs:line 216)
`
List of websites (following list - it is just example. errors occur on any links):
```c#
  string[] websites = new[]
  {
      "https://www.codeproject.com/",
      "https://github.com/",
      "https://visualstudio.microsoft.com/",
      "https://code.visualstudio.com/",
      "https://www.spyder-ide.org/",
      "https://nodejs.org/en/",
      "https://en.wikipedia.org/",
      "https://www.jetbrains.com/rider/",
      "https://www.jetbrains.com/pycharm/",
      "https://www.anaconda.com/",
      "https://www.python.org/",
  };
```

**VERSIONS:**
Chrome browser: Version 106.0.5249.119 (Official Build) (64-bit)
Chrome was started with args:  --remote-debugging-port=9222
OS: windows 11 home 64bit
RAM: 32gb
PuppeteerSharp : 8.0.0
.NET framework 4.8

### PR DESCRIPTION
**Pull request by kblok**
closes #2031

### CODE REVIEW

File: lib/PuppeteerSharp/Browser.cs [hunk 1]
```diff
@@ -352,98 +441,44 @@ private async void Connection_Disconnected(object sender, EventArgs e)
...
+            Connection.Disconnected += Connection_Disconnected;
+            TargetManager.TargetAvailable += TargetManager_TargetAvailable;
+            TargetManager.TargetGone += TargetManager_TargetGone;
+            TargetManager.TargetChanged += TargetManager_TargetChanged;
+            TargetManager.TargetDiscovered += TargetManager_TargetDiscovered;
+            return TargetManager.InitializeAsync();
         }
 
-        private void ChangeTargetInfo(TargetCreatedResponse e)
+        private void Detach()
```
Comment (jnyrup) [inline lib/PuppeteerSharp/Browser.cs:454]: I noticed that `Detach` is not referenced anywhere.
Where should it be invoked from?

File: lib/PuppeteerSharp/Browser.cs [hunk 2]
```diff
@@ -22,34 +22,61 @@ public class Browser : IBrowser
         private readonly ConcurrentDictionary<string, BrowserContext> _contexts;
         private readonly ILogger<Browser> _logger;
         private readonly Func<TargetInfo, bool> _targetFilterCallback;
+        private readonly Func<TargetInfo, bool> _isPageTargetFunc;
         private readonly BrowserContext _defaultContext;
         private readonly CustomQueriesManager _customQueriesManager = new();
-
+        private readonly Func<Task> _closeCallback;
```
Comment (jnyrup) [inline lib/PuppeteerSharp/Browser.cs:28]: This callback is also not invoked anywhere.

File: lib/PuppeteerSharp/Browser.cs (hunk 1 above)
Comment (kblok) [reply lib/PuppeteerSharp/Browser.cs:454]: True. I ask for confirmation [upstream](https://github.com/puppeteer/puppeteer/pull/9782)

File: lib/PuppeteerSharp/Browser.cs (hunk 2 above)
Comment (kblok) [reply lib/PuppeteerSharp/Browser.cs:28]: Ticket created here https://github.com/hardkoded/puppeteer-sharp/issues/2100 :) Thanks!

**Pull request merged**